File: Python/#922_sortArrayByParityII.py
class Solution:
    def sortArrayByParityII(self, A: List[int]) -> List[int]:
        i = 0
        j = 1
        n = len(A)
        while i<n and j<n:
            if A[i]%2 == 0:
                i+=2
            elif A[j]%2 == 1:
                j+=2
            else:
                A[i],A[j] = A[j],A[i]
                i+=2
                j+=2
        return A

File: Java/#922_sortArrayByParityII.java
class Solution {
    public int[] sortArrayByParityII(int[] A) {
        int i=0;
        int j=1;
        int n=A.length;
        while(i<n && j<n){
            if(A[i]%2==0){
                i+=2;
            }else if(A[j]%2==1){
                j+=2;
            }else{
                int temp=A[i];
                A[i]=A[j];
                A[j]=temp;
            }
        }
        return A;
    }
}
